### PR TITLE
Add a monthly check that everything is up-to-date

### PR DESCRIPTION
**Pull request by DanTup**
@devoncarew this script runs the existing bot script and then checks there are no Git differences in the `resources` folder, excluding `version.json`. The intention was to only pick up "real" changes and not trigger on every beta.

However, I was mostly thinking about colours/icons and now realise that `catalog/widgets.json` will change on every beta (because it has the version number in it). We could also exclude that file (although it means we could still miss some changes if there are no icons/colours changed in a release), or just update it on each beta.

It also occurs to me that in theory we could actually just have the bot commit the changes from running this (or perhaps automatically open a PR, which might be slightly better permission-wise).

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- N/A I listed at least one issue that this PR fixes in the description above.
- N/A I updated/added relevant documentation (doc comments with `///`).
- N/A I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by devoncarew — APPROVED** [review]
lgtm w/ the indicate change

There are some github actions that allow you to create issues; you could use that and a condition in the workflow file to create an issue when this script fails. I'm not sure I want to bring in new action deps just for that use however. This script as is improves the state of affairs; we can try it for a while and see if we want any changes down the road.

File: .github/workflows/check_up_to_date.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+on:
+  schedule:
+    # 10am, 1st of every month
+    - cron: '0 10 1 * *'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (DanTup) [reply .github/workflows/check_up_to_date.yaml]: Good catch - this was copied from the other file in this folder, so I've also updated that to v3. I've also added a `.github/dependabot.yml` to auto-update with basic/default settings. There are many things that can be customised but I'm not sure exactly what's best so feel free to tweak. It's not clear to me whether using a hash would pin it (and never update) though? Docs are here:

https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file

**Review by devoncarew — APPROVED** [review]

File: .github/workflows/check_up_to_date.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (devoncarew) [reply .github/workflows/check_up_to_date.yaml]: Thanks! I wasn't intending for you have have to add a dependabot config in this PR but thanks for doing so.

The dependabot frequency looks good (I've used weekly and monthly). You'll get security related version bumps out of band of whatever frequency you set up.

Using a semantic tag (`v3`, `v1.3`, ...) is nice and convenient, but by deping in an action you're giving another repo rights to perform actions here (create issues, PRs,  perform other actions). Tags can be moved after the fact by other repos, so whatever version you thought you were using or had vetted can be changed out from under you. Using a sha points to an exact set of source, and dependabot has support for them (knows that a sha is a version; updates you to the newer version as necessary, but writes the new version back in its sha form).

[the remark this answers is not in the document]
Comment (DanTup) [reply .github/workflows/check_up_to_date.yaml]: Ah nice, that all makes sense. Lemme know if you want me to change to a hash (or feel free to change if it's easier for you to just edit directly).

I'm also not sure who (if anyone) is notified if this new scheduled action fails. I'm happy to be notified if that's possible. If not, perhaps it makes sense to have it automatically open a PR (at least once we think it's working reasonable well).

**Pull request merged**